src/shiftedCompositeNormL2.jl

```diff
@@ -0,0 +1,223 @@
+export ShiftedCompositeNormL2
+@doc raw"""
+    ShiftedCompositeNormL2(h, c!, J!, A, b)
+
+Returns the shift of a function c composed with the ``\ell_{2}`` norm (see CompositeNormL2.jl).
+Here, c is linearized i.e, ``c(x+s) \approx c(x) + J(x)s``. 
+```math
+f(s) = λ \|c(x) + J(x)s\|_2
+```
+where ``\lambda > 0``. c! and J! should be functions
+```math
+\begin{aligned}
+&c(x) : \mathbb{R}^n \xrightarrow[]{} \mathbb{R}^m \\
+&J(x) : \mathbb{R}^n \xrightarrow[]{} \mathbb{R}^{m\times n}
+\end{aligned}
+```
+such that J is the Jacobian of c. A and b should respectively be a matrix and a vector which can respectively store the values of J and c.
+A is expected to be sparse, c and J should have signatures
+c!(b <: AbstractVector{Real}, xk <: AbstractVector{Real})
+J!(A <: AbstractSparseMatrixCOO{Real,Integer}, xk <: AbstractVector{Real})
+"""
+mutable struct ShiftedCompositeNormL2{
+  R <: Real,
+  V0 <: Function,
+  V1 <: Function,
+  V2 <: AbstractMatrix{R},
+  V3 <: AbstractVector{R},
+} <: ShiftedCompositeProximableFunction
+  h::NormL2{R}
+  c!::V0
+  J!::V1
+  A::V2
+  Aᵧ::V2
+  b::V3
+  g::V3
+  res::V3
+  sol::V3
+  dsol::V3
+  p::V3
+  dp
+  function ShiftedCompositeNormL2(
+    λ::R,
+    c!::Function,
+    J!::Function,
+    A::AbstractMatrix{R},
+    b::AbstractVector{R},
+  ) where {R <: Real}
+    p = similar(b, A.n + A.m)
+    dp = similar(b, A.n + A.m)
+    g = similar(b)
+    res = similar(b)
+    sol = similar(b)
+    dsol = similar(b)
+    Aᵧ = SparseMatrixCOO(A.m,A.n + A.m, similar(A.rows,length(A.rows) + A.m), similar(A.cols, length(A.cols) + A.m), similar(A.vals,length(A.vals) + A.m))
+    if length(b) != size(A,1)
+      error("ShiftedCompositeNormL2: Wrong input dimensions, there should be as many constraints as rows in the Jacobian")
+    end
+    new{R,typeof(c!),typeof(J!),typeof(A),typeof(b)}(NormL2(λ),c!,J!,A,Aᵧ,b,g,res,sol,dsol,p,dp)
+  end
+end
+
+
+shifted(λ::R, c!::Function, J!::Function, A::AbstractMatrix{R}, b::AbstractVector{R}, xk :: AbstractVector{R}) where {R <: Real} = begin
+  c!(b,xk)
+  J!(A,xk)
+  ShiftedCompositeNormL2(λ,c!,J!,A,b)
+end
+
+shifted(
+  ψ::ShiftedCompositeNormL2{R, V0, V1, V2, V3},
+  xk::AbstractVector{R},
+) where {R <: Real, V0 <: Function, V1 <: Function, V2 <: AbstractMatrix{R}, V3<: AbstractVector{R}} = begin
+  b = similar(ψ.b)
+  ψ.c!(b,xk)
+  A = similar(ψ.A)
+  ψ.J!(A,xk)
+  ShiftedCompositeNormL2(ψ.h.lambda, ψ.c!, ψ.J!, A, b)
+end
+ 
+shifted(
+  ψ::CompositeNormL2{R, V0, V1, V2, V3},
+  xk::AbstractVector{R}
+) where {R <: Real, V0 <: Function, V1 <: Function, V2 <: AbstractMatrix{R}, V3 <: AbstractVector{R}} = begin
+  b = similar(ψ.b)
+  ψ.c!(b,xk)
+  A = copy(ψ.A)
```

Comment:
You are right. I think it was because we did not have a `similar` function for `SparseMatrixCOO` when I wrote this. This means the compatibility of SparseMatrixCOO in `Projet.toml` should be updated as well to match the PR [SparseMatricesCOO.jl#40](https://github.com/JuliaSmoothOptimizers/SparseMatricesCOO.jl/pull/40)